internal/martian/header/via_modifier.go

```diff
@@ -49,42 +55,36 @@ func NewViaModifier(requestedBy string) *ViaModifier {
 //
 // http://tools.ietf.org/html/draft-ietf-httpbis-p1-messaging-14#section-9.9
 func (m *ViaModifier) ModifyRequest(req *http.Request) error {
-	via := fmt.Sprintf("%d.%d %s-%s", req.ProtoMajor, req.ProtoMinor, m.requestedBy, m.boundary)
+	via := req.Header.Get("Via")
+
+	l := m.len
+	if via != "" {
+		l += len(via) + 2
+	}
+
+	var sb strings.Builder
+	sb.Grow(l)
 
-	if v := req.Header.Get("Via"); v != "" {
-		if m.hasLoop(v) {
+	if via != "" {
+		if strings.Contains(via, m.requestedBy+"-"+m.boundary) {
```

Comment:
How about precomputing `m.requestedBy+"-"+m.boundary` instead of concatenating on every modify?

Reply:
Sure.